### PR TITLE
Add tests on notify jobs to prevent failures

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Add new tests on notify jobs to validate their execution.

### Motivation
Pipeline stability

### Additional Notes
This PR contains a first commit which is an aside change to set `datadog-agent` as default repo when calling github/gitlab API. Can be reviewed separately

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

File: tasks/notify.py [hunk 1]
```diff
@@ -231,30 +231,30 @@ def check_consistent_failures(ctx):
     send_notification(alert_jobs)
 
     # Upload document
-    with open("job_executions.json", "w") as f:
+    with open(job_failures_file, "w") as f:
         json.dump(job_executions, f)
-    ctx.run(f"{AWS_S3_CP_CMD} {JOB_FAILURES_FILE} {S3_CI_BUCKET_URL}/{JOB_FAILURES_FILE} ", hide="stdout")
+    ctx.run(f"{AWS_S3_CP_CMD} {job_failures_file} {S3_CI_BUCKET_URL}/{JOB_FAILURES_FILE} ", hide="stdout")
```
Comment (spencergilbert) [inline tasks/notify.py:236]: **❓ question:** ‏ should this have been updated as well?
```suggestion
    ctx.run(f"{AWS_S3_CP_CMD} {job_failures_file} {S3_CI_BUCKET_URL}/{job_failures_file} ", hide="stdout")
```

File: tasks/notify.py [hunk 2]
```diff
@@ -231,30 +231,30 @@ def check_consistent_failures(ctx):
...
 
-def retrieve_job_executions(ctx):
+def retrieve_job_executions(ctx, job_failures_file):
     """
     Retrieve the stored document in aws s3, or create it
     """
     try:
-        ctx.run(f"{AWS_S3_CP_CMD}  {S3_CI_BUCKET_URL}/{JOB_FAILURES_FILE} {JOB_FAILURES_FILE}", hide=True)
-        with open(JOB_FAILURES_FILE) as f:
+        ctx.run(f"{AWS_S3_CP_CMD}  {S3_CI_BUCKET_URL}/{job_failures_file} {JOB_FAILURES_FILE}", hide=True)
```
Comment (spencergilbert) [inline tasks/notify.py:244]: Same as above.
```suggestion
        ctx.run(f"{AWS_S3_CP_CMD}  {S3_CI_BUCKET_URL}/{job_failures_file} {job_failures_file}", hide=True)
```

Comment (chouetz) [reply tasks/notify.py:244]: Thanks, I forgot the `/g` in the sed :D

**Review by spencergilbert — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

File: tasks/unit-tests/notify_tests.py [hunk 3]
```diff
@@ -8,33 +10,76 @@
...
+        self.job_executions = notify.create_initial_job_executions(self.job_file)
 
     def tearDown(self) -> None:
-        pathlib.Path(notify.JOB_FAILURES_FILE).unlink(missing_ok=True)
+        pathlib.Path(self.job_file).unlink(missing_ok=True)
 
     def test_retrieved(self):
         ctx = MockContext(run=Result("test"))
-        j = notify.retrieve_job_executions(ctx)
+        j = notify.retrieve_job_executions(ctx, "job_executions.json")
```
Comment (KevinFairise2) [inline tasks/unit-tests/notify_tests.py:65]:
```suggestion
        j = notify.retrieve_job_executions(ctx, self.job_file)
```

**Review by KevinFairise2 — APPROVED** [review]
LGTM!

**Pull request merged**